### PR TITLE
Ys main branch

### PR DESCRIPTION
**Pull request by YSpiegel**
So how do we do it

### CODE REVIEW

File: Interface.py [hunk 1]
```diff
@@ -7,24 +7,149 @@
 window.geometry("600x400")
 window.title("Teachers Birthdays by Yoav Spiegel and Ido Kedem")
 
-# show today's date on the top abd add a sub-title
+
+def format_date(input_date):
+    formatted = input_date.strftime("%x")
```
Comment (IdoKedem) [inline Interface.py]: chnage %x to %d/%m

File: Interface.py [hunk 2]
```diff
@@ -7,24 +7,149 @@
...
-for date in general_BDays:
-    if date == today[0:5]:
+
+for date in general_BDays:  # show only teachers born today
+    if date == today:
         for teacher in general_BDays[date]:
             teachers_label_text += teacher + "\n"
-teachers_label = tk.Label(window, text=teachers_label_text, font=("arial", 10))
+
+teachers_label = tk.Label(window, text=teachers_label_text, font=("arial", 12))
```
Comment (IdoKedem) [inline Interface.py:42]: maybe add a line break inside brackets, your choice

File: Interface.py [hunk 3]
```diff
@@ -7,24 +7,149 @@
...
 teachers_label.pack()
 
+
+# Second part: date buttons
+
+def update_to_date():
+    '''
```
Comment (IdoKedem) [inline Interface.py]: docstrings are usually written with 3 double commas: """ and not '''

File: Interface.py [hunk 4]
```diff
@@ -7,24 +7,149 @@
...
+
+    Uses the current date to show the correct messages in every label concerning it
```
Comment (IdoKedem) [inline Interface.py:50]: maybe consider mentioning the func returns none?
:return: None

Comment (IdoKedem) [reply Interface.py:50]: can also do with "-> None" after the func's arguments braces

File: Interface.py [hunk 5]
```diff
@@ -7,24 +7,149 @@
...
+
+        for date in general_BDays:
+            if date == formatted_date[0:5]:
+                for teacher in general_BDays[date]:
+                    teachers_label_text += teacher + "\n"
+
+    teachers_label.configure(text=teachers_label_text)
+
+
+def minus_day():  # go one day backwards and update
```
Comment (IdoKedem) [inline Interface.py:79]: are you using """ or # to explain functions?

File: Interface.py [hunk 6]
```diff
@@ -7,24 +7,149 @@
...
+
+
+def mode_change():  # update state to present by date and retrieve plus/minus day buttons (kill button too)
+    update_to_date()
+    return_to_date_mode.place_forget()
+    plus_day.place(x=210, y=300)
+    minus_day.place(x=335, y=300)
+
+
+return_to_date_mode = tk.Button(window, text="Return to date mode", command=mode_change)
```
Comment (IdoKedem) [inline Interface.py:105]: maybe add a line break inside brackets, your choice

File: Interface.py [hunk 7]
```diff
@@ -7,24 +7,149 @@
...
+def plus_day():  # go one day forward and update
+    global current_date
+    current_date += dt.timedelta(days=1)
+    update_to_date()
+
+
+plus_day = tk.Button(window, text="⮝", width=4, height=2, command=plus_day)
+plus_day.place(x=210, y=300)
+
+minus_day = tk.Button(window, text="⮟", width=4, height=2, command=minus_day)
```
Comment (IdoKedem) [inline Interface.py:94]: maybe add a line break inside brackets, your choice

File: Interface.py [hunk 8]
```diff
@@ -7,24 +7,149 @@
...
+    update_to_date()
+
+
+def plus_day():  # go one day forward and update
+    global current_date
+    current_date += dt.timedelta(days=1)
+    update_to_date()
+
+
+plus_day = tk.Button(window, text="⮝", width=4, height=2, command=plus_day)
```
Comment (IdoKedem) [inline Interface.py:91]: maybe add a line break inside brackets, your choice

File: Interface.py [hunk 9]
```diff
@@ -7,24 +7,149 @@
...
+# Third part: month buttons
+
+def update_to_month(month, month_name):
+    '''
+
+    Used by the 12 month buttons
+    Updates the teachers list according to the month parameters
+    Deletes plus/minus buttons and adds mode change button
+    :param month: two digit number to sort dates by
+    :param month_name: the name that shall be presented on the main label
```
Comment (IdoKedem) [inline Interface.py:117]: amazing docstring!

**Review by IdoKedem — COMMENTED** [review]
great work! left some CR comments for you

File: Interface.py (hunk 1 above)
Comment (YSpiegel) [reply Interface.py]: done

File: Interface.py (hunk 3 above)
Comment (YSpiegel) [reply Interface.py]: done

File: Interface.py (hunk 4 above)
Comment (YSpiegel) [reply Interface.py:50]: Used the first one

File: Interface.py (hunk 5 above)
Comment (YSpiegel) [reply Interface.py:79]: """ if it's a complex one, # if it's a simple one

File: Interface.py (hunk 7 above)
Comment (YSpiegel) [reply Interface.py:94]: Like it that way idk

File: Interface.py (hunk 2 above)
Comment (YSpiegel) [reply Interface.py:42]: Like it like that

File: Interface.py (hunk 6 above)
Comment (YSpiegel) [reply Interface.py:105]: Read other comments of mine

File: Interface.py (hunk 8 above)
Comment (YSpiegel) [reply Interface.py:91]: You know it

File: Interface.py (hunk 9 above)
Comment (YSpiegel) [reply Interface.py:117]: GG EZ tnx

File: Interface.py [hunk 10]
```diff
@@ -9,27 +9,153 @@
 window.geometry("600x400")
 window.title("Teachers Birthdays by Yoav Spiegel and Ido Kedem")
 
-# show today's date on the top abd add a sub-title
+
+def format_date(input_date):
+    formatted = input_date.strftime("%d/%m")
+    return formatted[3:6] + formatted[0:2]
```
Comment (IdoKedem) [inline Interface.py:15]: can straight up return input_date.strftime("%d/%m")

File: Interface.py [hunk 11]
```diff
@@ -9,27 +9,153 @@
...
-today_msg = tk.Label(window, text="Today: " + today, font=("arial", 20))
-today_msg.pack()
-who_has_bd = tk.Label(window, text="Teachers who have a birthday today: ", font=("arial", 12), pady=10)
+global current_date
+current_date = random_date.today()
+date_label = tk.Label(window, text=format_date(current_date), font=("arial", 20))
+date_label.place(x=255, y=300)
+# tomorrow = current_date + dt.timedelta(days=1)
+today = format_date(current_date)
+who_has_bd = tk.Label(window, text="Teachers who have a birthday today: ", font=("arial", 20), pady=10)  # default msg
```
Comment (IdoKedem) [inline Interface.py:27]: for the love of god

**Pull request merged**